src/Types/point_types.jl

```diff
@@ -52,14 +55,12 @@ end
 @inline getindex(point_types::PointTypes{T, N, S}, I::Vararg{Int, N}) where {T, N, S} = getindex(point_types.data, I...)
 @inline getindex(point_types::PointTypes{T, N, S}, i::Int) where {T, N, S} = getindex(point_types.data, i)
 @inline getindex(point_types::PointTypes{T, N, S}, s::Symbol) where {T, N, S} = getindex(point_types.grid, s)
+@inline getindex(point_types::PointTypes{T, N, S}, pt::AbstractCoordinatePoint{T}) where {T, N, S} = point_types.data[find_closest_gridpoint(pt, point_types.grid)...]
 
 function in(pt::AbstractCoordinatePoint{T}, point_types::PointTypes{T, 3, S})::Bool where {T <: SSDFloat, S}
     cpt = _convert_point(pt, S)
-    grid::Grid{T, 3, S} = point_types.grid
-    i1::Int = searchsortednearest(grid.axes[1].ticks, cpt[1])
-    i2::Int = searchsortednearest(grid.axes[2].ticks, cpt[2])
-    i3::Int = searchsortednearest(grid.axes[3].ticks, cpt[3])
-    return point_types.data[i1, i2, i3] & pn_junction_bit > 0
+    point_type::PointType = point_types[cpt]
+    return (point_type & pn_junction_bit > 0) && (point_type & surface_bit == 0)

```

Comment:
Here, only points are "in" the `PointTypes if their `pn_junction_bit` is 1, but the `surface_bit` is 0.

Reply:
We could change `surface_bit` to something like `bulk_bit` so that we only have to check one bit here.

Reply:
That is a very good proposal, I changed this.